iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractTerminal.java

```diff
@@ -198,12 +203,12 @@ public boolean disconnect(Predicate<Switch> isSwitchOpenable) {
         if (removed) {
             throw new PowsyblException(UNMODIFIABLE_REMOVED_EQUIPMENT + connectable.id);
         }
+        int variantIndex = getVariantManagerHolder().getVariantIndex();
+        String variantId = getVariantManagerHolder().getVariantManager().getVariantId(variantIndex);
+        boolean disconnectedBefore = !isConnected();
+        connectable.notifyUpdate("beginDisconnect", variantId, disconnectedBefore, null);
         boolean disconnected = voltageLevel.disconnect(this, isSwitchOpenable);
-        if (disconnected) {
-            int variantIndex = getVariantManagerHolder().getVariantIndex();
-            String variantId = getVariantManagerHolder().getVariantManager().getVariantId(variantIndex);
-            connectable.notifyUpdate("connection", variantId, true, false);
-        }
+        connectable.notifyUpdate("endDisconnect", variantId, !disconnected, disconnected);
```

Comment:
Same comment as for `"endConnect"` notification.

We could either always use `true` as `newValue`:
```suggestion
        connectable.notifyUpdate("endDisconnect", variantId, disconnectedBefore, true);
```

or use `null` when the terminal was already disconnected:
```suggestion
        connectable.notifyUpdate("endDisconnect", variantId, disconnectedBefore, disconnected ? true : null);
```